mongorestore/restore.go

```diff
@@ -530,7 +531,23 @@ func (restore *MongoRestore) RestoreCollectionToDB(dbName, colName string,
 				watchProgressor.Set(file.Pos())
 			}
 			// flush the remaining docs
-			result.combineWith(NewResultFromBulkResult(bulk.Flush()))
+			bwResult, bwErr := bulk.TryFlush()
+			defer bulk.ResetBulk()
+
+			if db.TimeseriesBucketNeedsMixedSchema(bwErr) {
+				// Modify the timeseries collection and retry flushing the bulk writer.
+				logicalColName, nameErr := db.GetTimeseriesColNameFromBucket(colName)
+				if nameErr != nil {
+					resultChan <- result.withErr(nameErr)
+					return
+				}
+				if collModErr := restore.AllowMixedSchemaInTimeseriesBucket(dbName, logicalColName); collModErr != nil {
+					resultChan <- result.withErr(errors.Wrap(collModErr, "failed to handle mixed schema in a timeseries bucket"))
```

Comment:
```suggestion
					resultChan <- result.withErr(errors.Wrap(collModErr, "failed to enable mixed schema in a timeseries bucket"))
```

But actually, I think this wrapping should be done in the `AllowMixedSchemaInTimeseriesBucket` method, rather than here.